### PR TITLE
Clean up scripts with shellcheck

### PR DESCRIPTION
**Pull request by pta2002**
Should take care of #77290, but since there's a good chance I'll do a second pass through the scripts, I'm leaving this here to check with CI.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
I'm going to r? @pietroalbini in the hopes that you have time to take a look at this, I personally do not feel like I can dedicate bandwidth to being comfortable approving this

**Comment by pta2002** [comment]
That should be all changes necessary, as far as I know.

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh [hunk 1]
```diff
@@ -14,10 +14,10 @@ echo ERROR: An error was encountered with the build.
 cat /tmp/build.log
 exit 1
 "
-  trap "$on_err" ERR
+  trap '$on_err' ERR
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh:17]: This has different behavior, doesn't it? Currently it expands $on_err when `trap` is executed this; after it will expand $on_error when an error occurs.

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh [hunk 2]
```diff
@@ -31,35 +31,35 @@ exit 1
 # First up, build a cross-compiler
 git clone --depth=1 https://git.haiku-os.org/haiku
 git clone --depth=1 https://git.haiku-os.org/buildtools
-cd $BUILDTOOLS/jam
+cd "$BUILDTOOLS/jam"
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh:34]: Hmm, don't you also need to change lines 7 and 9 for this to do anything?

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh [hunk 3]
```diff
@@ -31,35 +31,35 @@ exit 1
...
+package extract -C $PACKAGE_ROOT $OUTPUT/objects/haiku/"$ARCH"/packaging/packages/haiku_devel.hpkg
 find $OUTPUT/download/ -name '*.hpkg' -exec package extract -C $PACKAGE_ROOT {} \;
 
 # Fix libgcc_s so we can link to it
 cd $PACKAGE_ROOT/develop/lib
 ln -s ../../lib/libgcc_s.so libgcc_s.so
 
 # Clean up
-rm -rf $BUILDTOOLS $HAIKU $OUTPUT/Jamfile $OUTPUT/attributes $OUTPUT/build \
+rm -rf "$BUILDTOOLS" "$HAIKU" $OUTPUT/Jamfile $OUTPUT/attributes $OUTPUT/build \
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh:64]:
```suggestion
rm -rf "$BUILDTOOLS" "$HAIKU" "$OUTPUT"/Jamfile "$OUTPUT"/attributes "$OUTPUT"/build \
```

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh [hunk 4]
```diff
@@ -31,35 +31,35 @@ exit 1
...
   $OUTPUT/build_packages $OUTPUT/download $OUTPUT/objects
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh:65]:
```suggestion
  "$OUTPUT"/build_packages "$OUTPUT"/download "$OUTPUT"/objects
```

File: src/ci/docker/host-x86_64/dist-riscv64-linux/crosstool-ng.sh [hunk 5]
```diff
@@ -6,7 +7,7 @@ curl -Lf $url | tar xzf -
 cd crosstool-ng-crosstool-ng-1.24.0
 ./bootstrap
 ./configure --prefix=/usr/local
-make -j$(nproc)
+make -j"$(nproc)"
```
Comment (jyn514) [inline src/ci/docker/host-x86_64/dist-riscv64-linux/crosstool-ng.sh:10]: There's no point in this; if `nproc` prints something that's not a number make will error anyway.

File: src/ci/docker/scripts/android-start-emulator.sh [hunk 6]
```diff
@@ -11,6 +11,6 @@ export SHELL=/bin/bash
 # classic engine the emulator starts with the current date and the tests run
 # fine. If another image is used, this need to be evaluated again.
 nohup nohup emulator @armeabi-v7a-18 \
-    -engine classic -no-window -partition-size 2047 0<&- &>/dev/null &
+    -engine classic -no-window -partition-size 2047 0<&- 2>&1 /dev/null &
```
Comment (jyn514) [inline src/ci/docker/scripts/android-start-emulator.sh:14]: This doesn't look right. Did you mean to redirect both?
```suggestion
    -engine classic -no-window -partition-size 2047 0<&- >/dev/null 2>&1 &
```
If so, why did you change it? It was the same before but shorter.

File: src/ci/init_repo.sh [hunk 7]
```diff
@@ -8,19 +8,17 @@ set -o errexit
...
+ci_dir=$(cd "$(dirname "$0")" && pwd)
 . "$ci_dir/shared.sh"
 
 REPO_DIR="$1"
 CACHE_DIR="$2"
 
-cache_src_dir="$CACHE_DIR/src"
-
-if [ ! -d "$REPO_DIR" -o ! -d "$REPO_DIR/.git" ]; then
+if [ ! -d "$REPO_DIR" ] || [ ! -d "$REPO_DIR/.git" ]; then
```
Comment (jyn514) [inline src/ci/init_repo.sh:17]: If you're going to avoid inconsistent behavior in `test`, let's be consistent about it ;)
```suggestion
if ! { [ -d "$REPO_DIR" ] || [ -d "$REPO_DIR/.git" ] }; then
```

File: src/ci/init_repo.sh [hunk 8]
```diff
@@ -41,26 +39,25 @@ function fetch_github_commit_archive {
...
-modules=($modules)
+read -ra modules <<< "$(git config --file .gitmodules --get-regexp '\.path$' | cut -d' ' -f2)"
 use_git=""
-urls="$(git config --file .gitmodules --get-regexp '\.url$' | cut -d' ' -f2)"
-urls=($urls)
-for i in ${!modules[@]}; do
-    module=${modules[$i]}
+read -ra urls <<< "$(git config --file .gitmodules --get-regexp '\.url$' | cut -d' ' -f2)"
+
+for i in "${!modules[@]}"; do
```
Comment (jyn514) [inline src/ci/init_repo.sh:53]: What is this doing? Why did you change it?

File: src/ci/shared.sh [hunk 9]
```diff
@@ -13,7 +13,9 @@ function retry {
   local n=1
   local max=5
   while true; do
-    "$@" && break || {
+    if "$@"; then
+      break
+    else
       if [[ $n -lt $max ]]; then
```
Comment (jyn514) [inline src/ci/shared.sh:19]:
```suggestion
    elif [[ $n -lt $max ]]; then
```
and also fix the indentation

File: src/test/run-make/git_clone_sha1.sh [hunk 10]
```diff
@@ -14,10 +14,10 @@ URL=$2
 SHA1=$3
```
Comment (jyn514) [inline src/test/run-make/git_clone_sha1.sh:14]: Github won't let me comment on lines that are unchanged, but please also change the shebang to `/usr/bin/env bash`.

File: src/test/run-make/thumb-none-qemu/script.sh [hunk 11]
```diff
@@ -1,19 +1,20 @@
...
 
 env | sort
-mkdir -p $WORK_DIR
-pushd $WORK_DIR
-    rm -rf $CRATE || echo OK
-    cp -a $HERE/example .
-    pushd $CRATE
+mkdir -p "$WORK_DIR"
+pushd "$WORK_DIR"
+    rm -rf "$CRATE" || echo OK
```
Comment (jyn514) [inline src/test/run-make/thumb-none-qemu/script.sh:9]: pre-existing, but this looks _super_ sketchy. @sekineh do you remember why you added this? Did you mean to write `&&` instead?

File: src/tools/clippy/.github/driver.sh [hunk 12]
```diff
@@ -28,14 +28,14 @@ diff normalized.stderr tests/ui/cstring.stderr
 
 
 # make sure "clippy-driver --rustc --arg" and "rustc --arg" behave the same
-SYSROOT=`rustc --print sysroot`
+SYSROOT=$(rustc --print sysroot)
```
Comment (jyn514) [inline src/tools/clippy/.github/driver.sh:31]:
```suggestion
SYSROOT="$(rustc --print sysroot)"
```

File: src/tools/clippy/.github/driver.sh [hunk 13]
```diff
@@ -28,14 +28,14 @@ diff normalized.stderr tests/ui/cstring.stderr
...
 diff <(LD_LIBRARY_PATH=${SYSROOT}/lib ./target/debug/clippy-driver --rustc --version --verbose) <(rustc --version --verbose)
```
Comment (jyn514) [inline src/tools/clippy/.github/driver.sh:32]:
```suggestion
diff <(LD_LIBRARY_PATH="${SYSROOT}"/lib ./target/debug/clippy-driver --rustc --version --verbose) <(rustc --version --verbose)
```

File: src/tools/clippy/.github/driver.sh [hunk 14]
```diff
@@ -28,14 +28,14 @@ diff normalized.stderr tests/ui/cstring.stderr
...
 
 
 echo "fn main() {}" > target/driver_test.rs
 # we can't run 2 rustcs on the same file at the same time
-CLIPPY=`LD_LIBRARY_PATH=${SYSROOT}/lib ./target/debug/clippy-driver ./target/driver_test.rs --rustc`
-RUSTC=`rustc ./target/driver_test.rs`
+CLIPPY=$(LD_LIBRARY_PATH=${SYSROOT}/lib ./target/debug/clippy-driver ./target/driver_test.rs --rustc)
```
Comment (jyn514) [inline src/tools/clippy/.github/driver.sh:37]:
```suggestion
CLIPPY=$(LD_LIBRARY_PATH="${SYSROOT}"/lib ./target/debug/clippy-driver ./target/driver_test.rs --rustc)
```

File: src/tools/clippy/.github/driver.sh [hunk 15]
```diff
@@ -28,14 +28,14 @@ diff normalized.stderr tests/ui/cstring.stderr
...
+RUSTC=$(rustc ./target/driver_test.rs)
 diff <($CLIPPY) <($RUSTC)
```
Comment (jyn514) [inline src/tools/clippy/.github/driver.sh:39]: This ... does not look right. Also this is a change to submodule so I'm not sure you should make it here, but clippy uses `subtree` so it might be special.

**Comment by jyn514** [comment]
```
Updating submodule src/doc/nomicon
error: could not lock config file .git/config: Read-only file system
error: could not lock config file .git/config: Read-only file system
fatal: Failed to register url for submodule path 'src/doc/nomicon'
```
I think this might be because you made the PR from the master branch? Not sure what's going on here.

**Comment by jyn514** [comment]
r? @jyn514 

I'm positive that pietro doesn't have time ;)

File: src/test/run-make/thumb-none-qemu/script.sh (hunk 11 above)
Comment (sekineh) [reply src/test/run-make/thumb-none-qemu/script.sh:9]: I meant `||` trying to catch the no existent path case.
But the entire `|| echo OK` seems not necessary because rm `-f` option already 'ignore nonexistent files and arguments' [[1]]

[1]: https://man7.org/linux/man-pages/man1/rm.1.html

Comment (sekineh) [reply src/test/run-make/thumb-none-qemu/script.sh:9]: `test -d "$CRATE" && rm -rf "$CRATE` might be better.

Comment (jyn514) [reply src/test/run-make/thumb-none-qemu/script.sh:9]: I think `rm -rf "$CRATE"` alone should be enough :) since like you said it ignores the files that are missing.

File: src/test/run-make/thumb-none-qemu/script.sh [hunk 16]
```diff
@@ -1,19 +1,20 @@
...
+    cp -a "$HERE"/example .
```
Comment (sekineh) [inline src/test/run-make/thumb-none-qemu/script.sh:10]:
```suggestion
    cp -a "$HERE"/"$CRATE" .
```

File: src/test/run-make/x86_64-fortanix-unknown-sgx-lvi/script.sh [hunk 17]
```diff
@@ -1,33 +1,35 @@
...
-    mkdir -p $WORK_DIR
-    pushd $WORK_DIR
-        rm -rf $CRATE
-        cp -a $TEST_DIR/enclave .
-        pushd $CRATE
-            echo ${WORK_DIR}
+    mkdir -p "$WORK_DIR"
+    pushd "$WORK_DIR"
+        rm -rf "$CRATE"
+        cp -a "$TEST_DIR"/enclave .
```
Comment (sekineh) [inline src/test/run-make/x86_64-fortanix-unknown-sgx-lvi/script.sh:10]:
```suggestion
        cp -a "$TEST_DIR"/"$CRATE" .
```

File: src/test/run-make/thumb-none-qemu/script.sh (hunk 11 above)
Comment (sekineh) [inline src/test/run-make/thumb-none-qemu/script.sh:9]:
```suggestion
    test -d "$CRATE" && rm -rf "$CRATE"
```

**Review by sekineh — COMMENTED** [review]
I added suggestions.

**Comment by pta2002** [comment]
> ```
> Updating submodule src/doc/nomicon
> error: could not lock config file .git/config: Read-only file system
> error: could not lock config file .git/config: Read-only file system
> fatal: Failed to register url for submodule path 'src/doc/nomicon'
> ```
> I think this might be because you made the PR from the master branch? Not sure what's going on here.

No idea either, the only thing I added to that file was #!/usr/bin/env bash, don't know how it could've made it fail.

**Comment by pta2002** [comment]
I'll get to work on these suggestions once I get home by the way, can probably finish this today still

File: src/tools/clippy/.github/driver.sh (hunk 15 above)
Comment (pta2002) [reply src/tools/clippy/.github/driver.sh:39]: Is there any difference between `` and $()? I changed it because shellcheck warned about it and as far as I know they do the same thing.

Comment (jyn514) [reply src/tools/clippy/.github/driver.sh:39]: Oh no your change is fine, I mean running eval on dynamically generated code. Normally I'd suggest to add quotes around the `$(...)`  but since I don't understand this code I'm afraid to change it.

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh (hunk 1 above)
Comment (pta2002) [reply src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh:17]: Yes, this has different behavior, however in this case it won't make a difference and it'd make ShellCheck shut up, but I can change it back if necessary.

File: src/ci/docker/scripts/android-start-emulator.sh (hunk 6 above)
Comment (pta2002) [reply src/ci/docker/scripts/android-start-emulator.sh:14]: You are right, it is what I meant. I changed it because POSIX sh does not support, &>, only bash.

File: src/ci/init_repo.sh (hunk 7 above)
Comment (pta2002) [reply src/ci/init_repo.sh:17]: Wouldn't this require the || to be changed to &&?

File: src/ci/init_repo.sh (hunk 8 above)
Comment (pta2002) [reply src/ci/init_repo.sh:53]: This is the "correct" way to extract a space-separated list into an array, there can be some issues with globbing using ($...). But now that I look at it I can probably do without using `cut` and just using read directly.

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh (hunk 1 above)
Comment (jyn514) [reply src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh:17]: Please change it back, yeah. I don't change the behavior just to satisfy the linter - it's possible the script is depending on the current behavior.

File: src/ci/init_repo.sh (hunk 7 above)
Comment (pta2002) [reply src/ci/init_repo.sh]: Hm I implemented this and it caused a syntax error, so I'll leave it as it was for now

File: src/ci/docker/scripts/android-start-emulator.sh [hunk 18]
```diff
@@ -11,6 +11,6 @@ export SHELL=/bin/bash
 # classic engine the emulator starts with the current date and the tests run
 # fine. If another image is used, this need to be evaluated again.
 nohup nohup emulator @armeabi-v7a-18 \
-    -engine classic -no-window -partition-size 2047 0<&- 2>&1 /dev/null &
+    -engine classic -no-window -partition-size 2047 0<&- 2>&1 > /dev/null &
```
Comment (jyn514) [inline src/ci/docker/scripts/android-start-emulator.sh:14]: This is not the same behavior: `2>&1` redirects to where stdout points _currently_, which is the terminal. You need to redirect _after_ you change stdout to get the previous behavior.
```suggestion
    -engine classic -no-window -partition-size 2047 0<&- >/dev/null 2>&1 &
```

File: src/ci/init_repo.sh [hunk 19]
```diff
@@ -14,7 +14,7 @@ ci_dir=$(cd "$(dirname "$0")" && pwd)
 REPO_DIR="$1"
 CACHE_DIR="$2"
 
-if [ ! -d "$REPO_DIR" ] || [ ! -d "$REPO_DIR/.git" ]; then
+if ! { [ -d "$REPO_DIR" ] && [ -d "$REPO_DIR/.git" ] }; then
```
Comment (jyn514) [inline src/ci/init_repo.sh]: Braces require semicolons.
```suggestion
if ! { [ -d "$REPO_DIR" ] && [ -d "$REPO_DIR/.git" ]; }; then
```

File: src/test/run-make/git_clone_sha1.sh [hunk 20]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -x
+#!/usr/bin/env bash
```
Comment (jyn514) [inline src/test/run-make/git_clone_sha1.sh:1]:
```suggestion
#!/usr/bin/env bash -x
```

File: src/ci/init_repo.sh (hunk 7 above)
Comment (jyn514) [reply src/ci/init_repo.sh:17]: You needed a semicolon before the `}`, I left it as a review comment.

**Comment by pta2002** [comment]
Alright, done - I still have no idea why CI is failing with .git being read only

File: src/ci/init_repo.sh [hunk 21]
```diff
@@ -8,19 +8,17 @@ set -o errexit
...
+ci_dir=$(cd "$(dirname "$0")" && pwd)
 . "$ci_dir/shared.sh"
 
 REPO_DIR="$1"
 CACHE_DIR="$2"
 
-cache_src_dir="$CACHE_DIR/src"
-
-if [ ! -d "$REPO_DIR" -o ! -d "$REPO_DIR/.git" ]; then
+if ! { [ -d "$REPO_DIR" ] && [ ! -d "$REPO_DIR/.git" ]; }; then
```
Comment (jyn514) [inline src/ci/init_repo.sh:17]: You have an extra `!` in here.
```suggestion
if ! { [ -d "$REPO_DIR" ] && [ -d "$REPO_DIR/.git" ]; }; then
```

Comment (pta2002) [reply src/ci/init_repo.sh:17]: Ooops, fixed

**Comment by jyn514** [comment]
```
error: could not lock config file .git/config: Read-only file system
error: could not lock config file .git/config: Read-only file system
```
Hmm, not sure what would cause this, let me take another look at the changes.

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh (hunk 2 above)
Comment (jyn514) [reply src/ci/docker/host-x86_64/disabled/dist-x86_64-haiku/build-toolchain.sh:34]: Hi @pta2002, you marked this as resolved but didn't change the lines. If they don't need to be changed, can you explain why?

File: src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh (hunk 1 above)
Comment (jyn514) [reply src/ci/docker/host-x86_64/disabled/dist-x86_64-dragonfly/build-toolchain.sh:17]: I think you missed this comment: please change `'$on_err'` back to `"$on_err"`.

File: src/ci/docker/run.sh [hunk 22]
```diff
@@ -152,28 +153,28 @@ else
...
-    args="$args --env AWS_SECRET_ACCESS_KEY"
+    args+=(--env SCCACHE_BUCKET)
+    args+=(--env SCCACHE_REGION)
+    args+=(--env AWS_ACCESS_KEY_ID)
+    args+=(--env AWS_SECRET_ACCESS_KEY)
 else
-    mkdir -p $HOME/.cache/sccache
-    args="$args --env SCCACHE_DIR=/sccache --volume $HOME/.cache/sccache:/sccache"
+    mkdir -p "$HOME/.cache/sccache"
+    args+=(--env "SCCACHE_DIR=/sccache" --volume "$HOME/.cache/sccache:/sccache")
```
Comment (jyn514) [inline src/ci/docker/run.sh:169]: This doesn't need quotes, it doesn't have variables.
```suggestion
    args+=(--env SCCACHE_DIR=/sccache --volume "$HOME/.cache/sccache:/sccache")
```

File: src/ci/docker/scripts/musl.sh [hunk 23]
```diff
@@ -32,11 +33,11 @@ if [ ! -d $MUSL ]; then
 fi
 
 cd $MUSL
-./configure --enable-optimize --enable-debug --disable-shared --prefix=/musl-$TAG $@
-if [ "$TAG" = "i586" -o "$TAG" = "i686" ]; then
-  hide_output make -j$(nproc) AR=ar RANLIB=ranlib
+./configure --enable-optimize --enable-debug --disable-shared --prefix=/musl-"$TAG" "$@"
+if [ "$TAG" = "i586" ] || [ "$TAG" = "i686" ]; then
```
Comment (jyn514) [inline src/ci/docker/scripts/musl.sh:37]:
```suggestion
if [ "$TAG" = i586 ] || [ "$TAG" = i686 ]; then
```

File: src/ci/init_repo.sh (hunk 8 above)
Comment (jyn514) [reply src/ci/init_repo.sh:53]: If I had to guess, this is what's causing CI to fail. I would revert the changes for now I think.

**Comment by pta2002** [comment]
Let's see if this works

**Comment by pta2002** [comment]
Seems like now it's failing due to not finding a cached build

**Comment by pta2002** [comment]
Hm from what I can tell, somewhere a .tar.gz got dropped in that URL, but where I have no idea, because I didn't remove any in my changes.

File: src/ci/run.sh [hunk 24]
```diff
@@ -126,13 +126,13 @@ trap datecheck EXIT
...
 if [ "$RUN_CHECK_WITH_PARALLEL_QUERIES" != "" ]; then
-  $SRC/configure --enable-parallel-compiler
+  "$SRC"/configure --enable-parallel-compiler
   CARGO_INCREMENTAL=0 $PYTHON ../x.py check
   rm -f config.toml
   rm -rf build
 fi
 
-$SRC/configure $RUST_CONFIGURE_ARGS
+"$SRC"/configure "$RUST_CONFIGURE_ARGS"
```
Comment (jyn514) [inline src/ci/run.sh:135]: This doesn't look correct. This used to be many different arguments, like `--set x --set y`; now it's one argument all at once. I think you should change it back to how it was before. If you want to silence the shellcheck warning you could trying using an array instead but I'd revert it for now just to make sure it works.

**Comment by pta2002** [comment]
Oh you're right, I didn't catch that! I'll revert it.

A sábado, 3/10/2020, 04:58, Joshua Nelson <notifications@github.com>
escreveu:

**Comment by pta2002** [comment]
Well that took a while, looks like it's fixed now though, at least CI has gone off to actually compile rust.

**Comment by jyn514** [comment]
> I personally do not feel like I can dedicate bandwidth to being comfortable approving this

@Mark-Simulacrum I'm reasonably happy with the current code, but I wouldn't stake my life on there being no regressions. How do you want to move forward?

**Comment by jyn514** [comment]
oh and while we're at it, let's try and catch more issues early

@bors try

**Comment by Mark-Simulacrum** [comment]
I don't have the time/bandwidth to invest in figuring out the shell script changes here, it's very error prone (as you have already seen) and I have no good ideas on avoiding regressions. IMO, it is questionable whether the improvements in this PR are worth it given the possibility of regressions and likelihood of reintroducing shellcheck-warned code over time.

Largely I just don't know that we get value out of this that outweighs the downsides.

That said, we have already branched beta, so it's as early in the cycle as it's going to get.

File: src/etc/cat-and-grep.sh [hunk 25]
```diff
@@ -59,7 +58,7 @@ if command -v "g${GREPPER}"; then
 fi
 
 LOG=$(mktemp -t cgrep.XXXXXX)
-trap "rm -f $LOG" EXIT
+trap 'rm -f $LOG' EXIT
```
Comment (jyn514) [inline src/etc/cat-and-grep.sh:61]: Please revert this change in behavior.

File: src/tools/clippy/.github/driver.sh [hunk 26]
```diff
@@ -28,14 +28,14 @@ diff normalized.stderr tests/ui/cstring.stderr
...
 
 # TODO: CLIPPY_CONF_DIR / CARGO_MANIFEST_DIR
```
Comment (Mark-Simulacrum) [inline src/tools/clippy/.github/driver.sh:41]: I would suggest that clippy *CI* changes are not done in this PR, as you really can't test them without running Clippy's CI. It should be fairly easily to port these to a Clippy PR though.

Comment (pta2002) [reply src/tools/clippy/.github/driver.sh:41]: That makes sense, I'll take these off and submit them to clippy separately
